### DISCUSSION TITLE
Extentions for shell scripting

### DISCUSSION THREAD

**Discussion by Susja** [microsoft/vscode-discussions#917] [Extension Development QnA] [answered]
Hello Folks,
I want to write and play a little bit with bash scripts.
I am looking for nice shell editor in VS Studio or VS Code.
I already installed the following: BATS, bash-snippets but I can't understand how to use it.
I am looking for some simple bash-oriented editor ... preferably with snippets.
Well .. I installed 'bash-snippets' extension but have no clue how to use it, same with BATS.
Could someone help me with it?

Thanks,

**Comment by isidorn** [comment, marked as answer]
You should just use VS Code and this extension should help https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck

**Reply by Susja** [reply]
@isidorn 
Appreciate your reply 
Last question: I already installed shellcheck extension on my Mac. How should I use it? I mean … should I invoke it explicitly? What it expects to handle: indentation, color legend, syntax issues or etc.?
Sorry for silly question… likely it doesn’t have a tutorial…but at least some heads-up of usage it. I never used it before 
Thanks
